tests/test_api_v1.py

```diff
@@ -2249,3 +2251,82 @@ def test_validate_package_manager_exclusivity(
             _validate_package_manager_exclusivity(pkg_managers, package_configs, mutually_exclusive)
     else:
         _validate_package_manager_exclusivity(pkg_managers, package_configs, mutually_exclusive)
+
+
+def test_get_content_manifests_by_requests(app, client, db, auth_env):
+    data = {
+        "repo": "https://github.com/release-engineering/dummy.git",
+        "ref": "c50b93a32df1c9d700e3e80996845bc2e13be848",
+        "pkg_managers": ["npm"],
+    }
+    with app.test_request_context(environ_base=auth_env):
+        request1 = Request.from_json(data)
+    request1.add_state(RequestStateMapping.complete.name, "Set complete directly for test")
+    db.session.add(request1)
+
+    data = {
+        "repo": "https://github.com/release-engineering/dummy-foo.git",
+        "ref": "450b93a32df1c9d700e3e80996845bc2e13be848",
+        "pkg_managers": ["npm"],
+    }
+    with app.test_request_context(environ_base=auth_env):
+        request2 = Request.from_json(data)
+    request2.add_state(RequestStateMapping.complete.name, "Set complete directly for test")
+    db.session.add(request1)
+
+    data = {
+        "repo": "https://github.com/release-engineering/dummy-bar.git",
+        "ref": "b50b93a32df1c9d700e3e80996845bc2e13be848",
+        "pkg_managers": ["npm"],
+    }
+    with app.test_request_context(environ_base=auth_env):
+        request3 = Request.from_json(data)
+    request3.add_state(RequestStateMapping.in_progress.name, "Set in_progress directly for test")
+    db.session.add(request3)
+
+    pkg1 = Package(name="pkg1", version="0.1", type="npm")
+    pkg2 = Package(name="pkg2", version="0.2", type="npm")
+    pkg3 = Package(name="pkg3", version="0.2", type="npm")
+    db.session.add(pkg1)
+    db.session.add(pkg2)
+    db.session.add(pkg3)
+
+    request1.add_package(pkg1)
+    request2.add_package(pkg2)
+    request3.add_package(pkg3)
+
+    db.session.commit()
+
+    test_data = [
```

Comment:
LGTM! I have only two suggestions about `data` & `test_data`: 
1. Can we move all `data` to `@pytest.mark.parametrize`?
2. If it's possible, we can move `test_data` to `@pytest.mark.parametrize` as well, and update only request ids. (perhaps it doesn't worth it).

Reply:
By moving to the `pytest.mark.parametrize`, the test_data cannot reference the `request.id` created in the test body.

Reply:
Yes, I see. It is why I mentioned that it may not worth moving `test_data`, and fill request ids later. Let's keep `test_data` in the test body :)